From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: SoftikLord <dimap9986@gmail.com>
Date: Thu, 24 Nov 2022 20:34:57 +0500
Subject: [PATCH] 
 Paper-Ability-to-get-Tile-Entities-from-a-chunk-without-snapshots


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftChunk.java b/src/main/java/org/bukkit/craftbukkit/CraftChunk.java
index 13e429dedb071f4d1650d30143b6bedda37ef7dc..43d136f865d06178862255a6c048e50bdba98be3 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftChunk.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftChunk.java
@@ -101,7 +101,12 @@ public class CraftChunk implements Chunk {
         return entities;
     }
 
+    // Paper start
     public BlockState[] getTileEntities() {
+        return getTileEntities(true);
+    }
+    public BlockState[] getTileEntities(boolean useSnapshot) {
+        // Paper end
         int index = 0;
         net.minecraft.server.Chunk chunk = getHandle();
 
@@ -113,7 +118,7 @@ public class CraftChunk implements Chunk {
             }
 
             BlockPosition position = (BlockPosition) obj;
-            entities[index++] = worldServer.getWorld().getBlockAt(position.getX(), position.getY(), position.getZ()).getState();
+            entities[index++] = worldServer.getWorld().getBlockAt(position.getX(), position.getY(), position.getZ()).getState(useSnapshot); // Paper
         }
 
         return entities;
